### PR TITLE
Bug 1538152 - "tell a story" about tasks in queue logging

### PR DESCRIPTION
**Pull request by djmitche**
This should allow us to search for `jsonPayload.Fields="<taskId>"` and find all of the interesting bits about a task.

Bugzilla Bug: [1538152](https://bugzilla.mozilla.org/show_bug.cgi?id=1538152)

### CODE REVIEW

**Comment by djmitche** [comment]
This also would let us search for workerGroup/workerId to see the claiming / reclaiming activity.  I thought about adding logging for declareWorker, too, but that's not used and will probably go away with worker-manager, so it's not worth the effort.

**Comment by djmitche** [comment]
OK, one last force-push and I'm hoping I'll get all green checkmarks.

File: libraries/monitor/README.md [hunk 1]
```diff
@@ -339,3 +339,16 @@ This function will:
  * shut down and flush monitoring
  * call `process.exit` on success or failure
 Note, then, that the function **does not return**.
+
+### Testing
+
+If the `mock` option is true, then all messges logged are available from the `messages` property on the MonitorManager, in the format `{Type, Fields, Logger}`.
+These messages can be cleared with the `reset()` method -- generally done in a Mocha `setup` function.
+This allows tests to assert that a message was logged correctly:
```
Comment (imbstack) [inline libraries/monitor/README.md]: :+1: good idea. I wasn't able to come up with a nice way to do this.

Comment (djmitche) [reply libraries/monitor/README.md]: Well, you did really -- it just wasn't documented and I implemented it in a worse way, then discovered your implementation and documented it :)

**Pull request merged**